esphome/components/preferences/syncer.h

```diff
@@ -8,15 +8,22 @@ namespace preferences {
 
 class IntervalSyncer : public Component {
  public:
-  void set_write_interval(uint32_t write_interval) { write_interval_ = write_interval; }
+  void set_write_interval(uint32_t write_interval) { this->write_interval_ = write_interval; }
   void setup() override {
-    set_interval(write_interval_, []() { global_preferences->sync(); });
+    if (this->write_interval_ != 0) {
```

Comment:
To respect the `never` value?
```suggestion
    if (this->write_interval_ != 0 && this->update_interval_ != 4294967295) {
```

Reply:
Not necessary, `set_interval()` already filters that out.

Reply:
:see_no_evil: :+1: